source/uwp/Renderer/lib/XamlBuilder.cpp

```diff
@@ -616,10 +622,272 @@ namespace AdaptiveCards { namespace XamlCardRenderer
         *uiShowCard = localUiShowCard.Detach();
     }
 
+    void SerializeTextInput(
+        Json::Value& jsonValue,
+        const char * idString,
+        InputItem* inputItem)
+    {
+        ComPtr<ITextBox> textBox;
+        THROW_IF_FAILED(inputItem->uiInputElement.As(&textBox));
+
+        HString text;
+        THROW_IF_FAILED(textBox->get_Text(text.GetAddressOf()));
+
+        std::string textString;
+        if (text.IsValid())
+        {
+            THROW_IF_FAILED(HStringToUTF8(text.Get(), textString));
+        }
+
+        jsonValue[idString] = textString.c_str();
+    }
```

Comment:
We spoke offline.  These don't belong in the public classes, because they're mostly pulling from the UI element rather than the adaptive card element.  Instead, expanded the InputItem struct into a class with a Serialize() method and moved all of this logic in there.

---
In reply to: [118372526](https://github.com/Microsoft/AdaptiveCards/pull/369#discussion_r118372526) [](ancestors = 118372526)